query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -382,9 +383,13 @@ private boolean isEquals(Set<String> values, Expression expression) {
         request = originalRequest;
       }
 
-      Entry<String, Params> pql =
+      Pair<Map<Integer,String>,Entry<String, Params>> pair =
```

Comment:
I think we don't have to change the runt signature of this function. The avg_rage function column index can be derived using the helper function by passing argument to it all `executionContext.getAllSelections()`